src/Sarif.Viewer.VisualStudio.Core/Tags/KeyEventAdornmentManager.cs

```diff
@@ -69,7 +66,9 @@ internal class KeyEventAdornmentManager
             this.layer = view.GetAdornmentLayer(nameof(KeyEventAdornmentManager));
 
             this.view = view;
-            this.view.LayoutChanged += this.OnLayoutChanged;
+            this.view.LayoutChanged += this.View_LayoutChanged;
+            this.view.GotAggregateFocus += this.View_GotAggregateFocus;
```

Comment:
Tried listening to `this.sarifEventService.NavigatedItemChanged`, this happens immediately after the `NavigatedItem` is set, at this time the text marker tags are not added to the text view yet. The text adornments depend on the text marker tags. 

`GotAggregateFocus` fires after the navigating to the error location, can be used to refresh the adornments after switching to another error.